### ISSUE TITLE
[Overlays] Push down blue "syncing" icon already during local discovery, not only at aboutToPropagate()

### ISSUE THREAD

**Issue by guruz** [owncloud/client#4731]
Because the discovery can take a long time.

I'm however not completely sure if this is needed. If the file manager sees a local change, will he not anyway re-request an icon which will then match the blue icon?

(But maybe we should do it because of possible remote changes by remote discovery? However how does this behave for reconcile)

FYI @ckamm @jturcotte @dragotin @ogoffart

**Comment by ckamm** [comment]
@guruz I think making files blue on file watcher notification would cover the local-modification case (#4730). Waiting for the discovery may take too long -- another folder could still be syncing for an hour. Updating the icons again once all trees have been compared and when we're about to start propagation seems like the right time to me.

**Comment by guruz** [comment]
@ckamm Good point. I'm moving this to 2.3 then because we could in theory still do it if there are _remote_ modifications (```EVAL```) for a file. Also a directory could turn to blue if we know that a _remote_ file will be created.

**Comment by ogoffart** [comment]
The shell is usualy notified when a file change and query the plugin for the icon. The thing is that the current implementation of the plugins keep a cache and do not clear the cache when this happens.  This could of course be changed.

**Comment by guruz** [comment]
@ogoffart Clear the cache when what exactly happens? And why a cache at all then?

**Comment by jturcotte** [comment]
If we send a SYNC, we need to keep track of those files and send a matching OK afterward if the propagation won't happen after all to absolutely make sure the icon is updated.

This would also only cover the case where the client isn't running when the file was changed, since otherwise the FileSystemWatcher will catch it. Since it's the local discovery that takes time, it's also only going to help if the file to be discovered is at the beginning of the list.

So not sure if this is worth it, I'll at least remove it from the milestone.
